openedx/core/djangoapps/django_comment_common/comment_client/utils.py

```diff
@@ -204,3 +218,46 @@ def check_forum_heartbeat():
             return 'forum', False, res.get('check', 'Forum heartbeat failed')
     except Exception as fail:
         return 'forum', False, str(fail)
+
+
+def check_service_enabled():
+    # To avoid dependency conflict
+    from openedx.core.djangoapps.django_comment_common.models import ForumsConfig
+
+    config = ForumsConfig.current()
+
+    if not config.enabled:
+        raise CommentClientMaintenanceError("service disabled")
+
+
+def handle_response(
```

Comment:
This function is preventing us from benefiting some of the niceties of a native API: explicit function signature, type checking, etc.

Instead, I propose that the forum API does not return status codes, but raises exceptions directly. Also, we should get rid of `metric_action` and raw requests.